packages/ai-vercel/src/interrupts/invokeTools.ts

```diff
@@ -166,16 +167,15 @@ const invokeToolsCoreMessage = async ({
             type: "tool-result",
             toolCallId: lastContent.toolCallId,
             toolName: lastContent.toolName,
-            result,
+            output: result,
           },
         ],
       });
     }
   } catch (err: any) {
-    throw new ToolExecutionError({
-      toolCallId: lastContent.toolCallId,
-      toolName: lastContent.toolName,
-      toolArgs: lastContent.args as JSONValue,
+    throw new AISDKError({
+      name: err.name,
+      message: err.message,
```

Comment:
@priley86 we should probably look into changing this. With v5 the recommended HITL approach is to *not* return an `execute` function. 

When I modified the SDK locally to get things working w/ v5 I changed this to return `undefined`. 

This results in the server sending the response back w/ `input-available`. The client, via useChat, then handles the user interaction and responds with `addToolResult` in order to kick the stream off again. 

It's a change from handling interrupts via errors vs the framework. Just a suggestion.

https://ai-sdk.dev/cookbook/next/human-in-the-loop#intercept-tool-call

Reply:
thanks @eatplaysleep  🙇 - i will take a look 👀